File: app/src/main/java/nz/co/codebros/quakesnz/detail/QuakeDetailModule.java
package nz.co.codebros.quakesnz.detail;

import android.arch.lifecycle.ViewModelProvider;
import android.arch.lifecycle.ViewModelProviders;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import nz.co.codebros.quakesnz.repository.FeatureRepository;

/**
 * Created by Leandro on 2/11/2017.
 */
@Module
public abstract class QuakeDetailModule {

    @Binds
    abstract QuakeDetailView quakeDetailView(QuakeDetailFragment fragment);

    @Provides
    static QuakeDetailViewModel.Factory factory(FeatureRepository repository) {
        return new QuakeDetailViewModel.Factory(repository);
    }

    @Provides
    static QuakeDetailViewModel quakeDetailViewModel(
            QuakeDetailFragment fragment,
            QuakeDetailViewModel.Factory factory
    ) {
        return ViewModelProviders.of(fragment, factory).get(QuakeDetailViewModel.class);
    }
}

File: app/src/main/java/nz/co/codebros/quakesnz/detail/QuakeDetailModule.kt
package nz.co.codebros.quakesnz.detail

import android.arch.lifecycle.ViewModelProviders
import dagger.Module
import dagger.Provides

/**
 * Created by Leandro on 16/02/2018.
 */
@Module
object QuakeDetailModule {

    @JvmStatic
    @Provides
    internal fun viewModel(
            fragment: QuakeDetailFragment,
            factory: QuakeDetailViewModel.Factory
    ) = ViewModelProviders.of(fragment, factory).get(QuakeDetailViewModel::class.java)
}